### PR TITLE
[webgpu] Add MatMulSmallOutputSizeProgram WGSL support

### PR DESCRIPTION
**Pull request by axinging**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5567)

### CODE REVIEW

**Comment by axinging** [comment]
@qjia7 @xhcao @haoyunfeix @gyagp , PTAL

**Review by qjia7 — APPROVED** [review]
@kainino0x @lina128 take a look, thanks.

File: tfjs-backend-webgpu/src/kernels/matmul_small_output_size_webgpu.ts [hunk 1]
```diff
@@ -107,29 +108,130 @@ export function makeMatMulSmallOutputSizeSource(): string {
...
+  var<workgroup> mm_Bsub2 : array<array<f32, ${tileBOuter}>, ${tileInner}>;
+
+  // If the output size is small for matrix multiplication, avoid to use vec4
+  // and handle some elements per thread to optimally utilize the ALU.
+  // Introduces two shared memory buffers, some logical threads could handle
+  // arithmetic operations and others handle IO operations between barrier api,
+  // makes ALUs and load/store units work simultaneously, could improves
+  // the performance.
+  ${getMainHeaderStringWgsl(workGroupSize)} {
+    ${getGlobalIndexStringWgsl(workGroupSize)}
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/matmul_small_output_size_webgpu.ts:129]: nit: Why is `getGlobalIndexStringWgsl` needed?

Comment (axinging) [reply tfjs-backend-webgpu/src/kernels/matmul_small_output_size_webgpu.ts]: You are right, it is not used, removed.

**Review by lina128 — APPROVED** [review]
LGTM

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5567#-:-MisFICR2hip45asBI-3:bb6ywmx)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @kainino0x and @qjia7)

**Pull request merged**